### PR TITLE
DCS-1877 handle temporary absence for different prison

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt [hunk 1]
```diff
@@ -126,15 +126,15 @@ class TemporaryAbsencesResource(
...
     path = [
-      "/prison/{prisonId}/temporary-absences/{prisonNumber}"
+      "/prison/{prisonId}/temporary-absences/{prisonNumber}", "/temporary-absences/{prisonNumber}"
     ]
   )
   fun getTemporaryAbsence(
-    @Schema(description = "Prison ID", example = "MDI", required = true)
-    @PathVariable prisonId: String,
+    @Schema(description = "Prison ID", example = "MDI", required = false)
+    @PathVariable prisonId: String?,
```
Comment (connormaglynn) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt:134]: Is this for backwards compatibility? I.e. we plan to remove `"/prison/{prisonId}/temporary-absences/{prisonNumber}"` once the UI has been updated - or something along those lines?

Comment (louisenorris) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/resources/TemporaryAbsencesResource.kt:134]: Yes exactly - for backwards compatibility until the frontend can be updated, then will go back and remove the old endpoint and prisonId :)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/temporaryabsences/TemporaryAbsenceService.kt [hunk 2]
```diff
@@ -10,20 +10,25 @@ import uk.gov.justice.digital.hmpps.welcometoprison.model.confirmedarrivals.Conf
...
-    getTemporaryAbsences(agencyId).find { it.prisonNumber == prisonNumber }
+  fun getTemporaryAbsence(prisonNumber: String): TemporaryAbsenceResponse {
+    val prisoner = prisonerSearchApiClient.getPrisoner(prisonNumber)
+      ?: throw NotFoundException("Could not find prisoner with prisonNumber: '$prisonNumber'")
+    return getTemporaryAbsences(prisoner.prisonId!!).find { it.prisonNumber == prisonNumber }
       ?: throw NotFoundException("Could not find temporary absence with prisonNumber: '$prisonNumber'")
+  }
 
-  fun getTemporaryAbsences(agencyId: String) = prisonApiClient.getTemporaryAbsences(agencyId).map {
+  fun getTemporaryAbsences(prisonId: String) = prisonApiClient.getTemporaryAbsences(prisonId).map {
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/temporaryabsences/TemporaryAbsenceService.kt:31]: 👍

**Review by andrewrlee — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/temporaryabsences/TemporaryAbsenceService.kt (hunk 2 above)
Comment (louisenorris) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/temporaryabsences/TemporaryAbsenceService.kt:31]: Thanks @michalnawrockidj :)

**Pull request merged**